core/src/main/java/org/apache/gravitino/cache/ReverseIndexRules.java

```diff
@@ -0,0 +1,147 @@
+                    switch (securableObject.type()) {
+                      case METALAKE:
+                        entityType = Entity.EntityType.METALAKE;
+                        namespace = NamespaceUtil.ofMetalake();
+                        break;
+                      case CATALOG:
+                        entityType = Entity.EntityType.CATALOG;
+                        namespace = NamespaceUtil.ofCatalog(roleEntity.namespace().level(0));
+                        break;
+                      case SCHEMA:
+                        entityType = Entity.EntityType.SCHEMA;
+                        Namespace nsSchema = Namespace.fromString(securableObject.parent());
+                        namespace =
+                            NamespaceUtil.ofSchema(
+                                roleEntity.namespace().level(0), nsSchema.level(0));
+                        break;
+                      case TABLE:
+                        entityType = Entity.EntityType.TABLE;
+                        Namespace nsTable = Namespace.fromString(securableObject.parent());
+                        namespace =
+                            NamespaceUtil.ofTable(
+                                roleEntity.namespace().level(0),
+                                nsTable.level(0),
+                                nsTable.level(1));
+                        break;
+                      case TOPIC:
+                        entityType = Entity.EntityType.TOPIC;
+                        Namespace nsTopic = Namespace.fromString(securableObject.parent());
+                        namespace =
+                            NamespaceUtil.ofTopic(
+                                roleEntity.namespace().level(0),
+                                nsTopic.level(0),
+                                nsTopic.level(1));
+                        break;
+                      case MODEL:
+                        entityType = Entity.EntityType.MODEL;
+                        Namespace nsModel = Namespace.fromString(securableObject.parent());
+                        namespace =
+                            NamespaceUtil.ofModel(
+                                roleEntity.namespace().level(0),
+                                nsModel.level(0),
+                                nsModel.level(1));
+                        break;
+                      case FILESET:
+                        entityType = Entity.EntityType.FILESET;
+                        Namespace nsFileset = Namespace.fromString(securableObject.parent());
+                        namespace =
+                            NamespaceUtil.ofFileset(
+                                roleEntity.namespace().level(0),
+                                nsFileset.level(0),
+                                nsFileset.level(1));
+                        break;
+                      default:
+                        LOG.info("Unprocessed securable object type: " + securableObject.type());
```

Comment:
I'm afraid that we may need to throw an exception here, or it will go to L40 and continue to run the rest of the code.